pkgs/build-support/compress-drv/default.nix

```diff
@@ -65,14 +73,22 @@ let
     ext: prog:
     assert validProg ext prog;
     ''
-      find -L $out -type f -regextype posix-extended -iregex '.*\.(${formatsPipe})' -print0 \
+      find -L $out -type f -regextype posix-extended -iregex '.*\.(${formatsPipe})' ${extraFindOperands} -print0 \
         | xargs -0 -P$NIX_BUILD_CORES -I{} ${prog}
     '';
-  formatsPipe = builtins.concatStringsSep "|" formats;
+  formatsPipe = lib.concatStringsSep "|" formats;
 in
-runCommand "${drv.name}-compressed" { } ''
-  mkdir $out
-  (cd $out; ${xorg.lndir}/bin/lndir ${drv})
+runCommand "${drv.name}-compressed"
+  (
+    {
+      nativeBuildInputs = [ xorg.lndir ];
+    }
+    // (lib.optionalAttrs (drv ? pname) { inherit (drv) pname; })
+    // (lib.optionalAttrs (drv ? version) { inherit (drv) version; })
```

Comment:
Can you help me understand why those 2 are needed for the wrappeijg derivation?

Reply:
There's an answer in the commit text of 5938ef4cff5a88997cd6d3421e35a3d14fed20e6:

> This is required to overwrite to use a compressed version of nextcloud in services.nextcloud.package because the module accesses version

Reply:
That is used here:

https://github.com/NixOS/nixpkgs/blob/b46afe538da7262ce53b9f32b4a5c4d715d79801/nixos/modules/services/web-apps/nextcloud.nix#L850-L854

and I do the following in my config.

```
services.nextcloud.package = pkgs.compressDrvWeb pkgs.nextcloud29 {
  extraFindOperands = ''-not -iregex ".*(\/apps\/.*\/l10n\/).*"'';
};
```